lib/screens/v2/candidate_generator/on_bus/departures.ex

```diff
@@ -80,6 +82,47 @@ defmodule Screens.V2.CandidateGenerator.Widgets.OnBus.Departures do
     Enum.filter(departures, &(&1.prediction.route.id != route_id))
   end
 
+  defp filter_duplicate_routes(departures) do
+    unique_departures =
+      Enum.uniq_by(departures, fn dep ->
+        {dep.prediction.route.id, dep.prediction.trip.direction_id}
```

Comment:
This is not quite the uniqueness criteria specified in the task, which is to use the definition of "destination" from the Real-time Destination States proposal. That consists of the Line (so prediction => route => line => ID) and the "representative headsign" of the trip (already implemented as `Screens.Trips.Trip.representative_headsign/1`).